modules/oe_translation_cdt/tests/src/Kernel/TranslationRequestBundleClassTest.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+declare(strict_types=1);
+
+use Drupal\oe_translation\Entity\TranslationRequest;
+use Drupal\oe_translation_cdt\TranslationRequestCdtInterface;
+use Drupal\oe_translation_remote\TranslationRequestRemoteInterface;
+use Drupal\Tests\oe_translation\Kernel\TranslationKernelTestBase;
+
+/**
+ * Tests the CDT bundle class for "Translation Request" entity.
+ *
+ * @group batch1
+ */
+class TranslationRequestBundleClassTest extends TranslationKernelTestBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  protected static $modules = [
+    'oe_translation_cdt',
+    'oe_translation_remote',
+  ];
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp(): void {
+    parent::setUp();
+    $this->installEntitySchema('oe_translation_request');
+    $this->installConfig([
+      'oe_translation_remote',
+      'oe_translation_cdt',
+    ]);
+  }
+
+  /**
+   * Tests the CDT bundle class setters and getters.
+   */
+  public function testCdtBundleClass(): void {
+    $request = TranslationRequest::create([
+      'bundle' => 'cdt',
+    ]);
+    assert($request instanceof TranslationRequestCdtInterface);
+
+    $request->setCdtId('test_cdt_id');
+    $this->assertEquals('test_cdt_id', $request->getCdtId());
+
+    $request->setRequestStatusFromCdt('COMP');
+    $this->assertEquals(TranslationRequestRemoteInterface::STATUS_REQUEST_TRANSLATED, $request->getRequestStatus());
+
+    $request->updateTargetLanguageStatusFromCdt('en', 'CMP');
+    $this->assertEquals(TranslationRequestRemoteInterface::STATUS_LANGUAGE_REVIEW, $request->getTargetLanguage('en')?->getStatus());
```

Comment:
```suggestion
    $this->assertEquals(TranslationRequestRemoteInterface::STATUS_LANGUAGE_REVIEW, $request->getTargetLanguage('en')->getStatus());
```
Is this a typo?

Reply:
I used the null-safe operator, because the exception thrown in case if `getTargetLanguage('en')` is `NULL` was misleading.

Reply:
And the other thing is that the PHPStan pipeline will fail without `?`, because of the type mismatch

```
------ ------------------------------------------------ 
  Line   TranslationRequestBundleClassTest.php           
 ------ ------------------------------------------------ 
  53     Cannot call method getStatus() on               
         Drupal\oe_translation\LanguageWithStatus|null.  
 ------ ------------------------------------------------ 
```